webapp/views/results-modal/script.js

```diff
@@ -1,16 +1,156 @@
+  }
+  get dataTimeMap() {
+    return this._data.filter(str => str.match(/^time/));
+  }
+  get dataQuery() {
+    return this._data.filter(str => str.match(/^query/));
+  }
+
+  remove(file) {
+    this.dataAll.splice(this.dataAll.indexOf(file), 1);
+  }
+
+  onClickTimeMap() {
+    const tbody = $('#results-table table tbody').empty();
+    $('#all-btn').removeClass('active');
+    $('#query-btn').removeClass('active');
+    $('#time-map-btn').addClass('active');
+
+    this.currentView = 'TIMEMAP';
+    this.dataTimeMap.forEach(file => {
+      this.appendItem(tbody, file);
+    });
+  }
+
+  onClickQuery() {
+    const tbody = $('#results-table table tbody').empty();
+    $('#all-btn').removeClass('active');
+    $('#query-btn').addClass('active');
+    $('#time-map-btn').removeClass('active');
+
+    this.currentView = 'QUERY';
+    this.dataQuery.forEach(file => {
+      this.appendItem(tbody, file);
+    });
+  }
+
+  onClickAll() {
+    const tbody = $('#results-table table tbody').empty();
+    $('#all-btn').addClass('active');
+    $('#query-btn').removeClass('active');
+    $('#time-map-btn').removeClass('active');
+
+    this.currentView = 'ALL';
+    this.dataAll.forEach(file => {
+      this.appendItem(tbody, file);
+    });
+  }
+
+  onClickDeleteAll() {
+    $('#delete-all-modal').show();
+  }
+
+  onClickDelete(btn) {
+    $('#delete-modal').show();
+
+    this.resultToDelete = btn.value;
+    $('#result-to-delete-span').text(resultModal.resultToDelete);
+  }
+
+  updateResults() {
+    get('/output', {}, res => new Promise(resolve => {
+      res.list.sort().reverse();
+      this.dataAll = res.list;
+
+      switch (this.currentView) {
+        case 'QUERY':
+          this.onClickQuery();
+          break;
+        case 'TIMEMAP':
+          this.onClickTimeMap();
+          break;
+        default:
+          this.onClickAll();
+          break;
+      }
+      resolve();
+    }));
+  }
+
+  appendItem(tbody, file) {
+    const m = file.match(/.*(\d{4})-(\d{2})-(\d{2})_(\d{4})\.pdf$/);
+    const date = new Date(m[1], m[2] - 1, m[3]);
+    tbody.append(`<tr>
+  <td>${date.toDateString()}</td>
+  <td><a href="/output/${file}" target="_blank">${file}</a></td>
+  <td><button type="button" class="btn btn-outline-danger" value="${file}" onclick="resultModal.onClickDelete(this)">Delete</button></td>
+  </tr>`);
+  }
+
+  hide() {
+    $('#results-modal').hide();
   }
 }
 
-// eslint-disable-next-line no-unused-vars
-function hideResults() {
-  $('#results-modal').hide();
+class DeleteModal {
+  constructor(resultModal) {
+    this.resultModal = resultModal;
+  }
+
+  hide() {
+    $('#delete-modal').hide();
+  }
+
+  onClickDelete() {
+    deleteMethod('/output', { file: this.resultModal.resultToDelete }, () => new Promise((resolve) => {
+      this.resultModal.remove(this.resultModal.resultToDelete);
+      resolve();
+    }));
+    this.resultModal.resultToDelete = '';
+    this.resultModal.updateResults();
```

Comment:
The correct way is to call `resultModal.updateResults` in the callback of the delete call (because you don't know yet if the delete operation will be successful). The `resultModal.remove` function is not needed.